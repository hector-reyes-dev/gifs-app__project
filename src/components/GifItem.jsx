import PropTypes from "prop-types";

export const GifItem = ({ title, src, id }) => {
  return (
    <div className="card">
      <img src={src} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  id: PropTypes.number,
};
