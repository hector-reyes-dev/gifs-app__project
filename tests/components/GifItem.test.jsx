import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("<GifItem /> tests", () => {
  const testTitle = "Testing title for GifItem";
  const testSrc = "https://testingsrc.com/";

  test("should match with snapshot", () => {
    const { container } = render(<GifItem title={testTitle} src={testSrc} />);
    expect(container).toMatchSnapshot();
  });

  test("should show img with url and alt", () => {
    render(<GifItem title={testTitle} src={testSrc} />);
    // screen.debug();
    // expect(screen.getByRole("img").src).toBe(testSrc);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(testSrc);
    expect(alt).toBe(testTitle);
  });

  test("should show title", () => {
    render(<GifItem title={testTitle} src={testSrc} />);
    expect(screen.getByText(testTitle)).toBeTruthy;
  });
});
