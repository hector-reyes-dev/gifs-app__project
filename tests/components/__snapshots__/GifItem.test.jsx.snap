// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem /> tests should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Testing title for GifItem"
      src="https://testingsrc.com/"
    />
    <p>
      Testing title for GifItem
    </p>
  </div>
</div>
`;
